### ISSUE TITLE
Not loading behind reverse proxy

### ISSUE THREAD

**Issue by sn0wcrashing** [6c65726f79/Transmissionic#803]
**This may be intended behaviour? If so please move to enhancement**

**Describe the bug**
WebUI not populating torrents list etc when used behind a reverse proxy. Remains with the animated loading circle.

**To Reproduce**
Place behind a reverse proxy (HA Proxy in my case) using a sub-domain (e.g. transmission.sub-domain.tld) and load in web browser. 

**Platform:**
Tested on MacOS Safari, Windows 10 Firefox. 

I haven't done extensive testing, just loaded this yesterday and it looks great (thank-you) but didn't load when using sub-domain address. Locally using IP address/port it loaded correctly.

**Comment by 6c65726f79** [comment]
Hi, thank you for reporting this issue. It's indeed a bug.

When you click on the pencil next to the server name, are the host and port the right ones? And are you using HTTPS?

**Comment by sn0wcrashing** [comment]
Yes they look correct, using https on port 443, with acme handling certificates on the reverse proxy. I can’t edit the authentication fields when editing the server.

**Comment by 6c65726f79** [comment]
Ok thanks for the information, I'll try to reproduce this bug on my side.

**Comment by 6c65726f79** [comment]
Hi, so I didn't managed to reproduce this bug. Could you open the dev tools (Ctrl+Shift+I) and show me if there's an error in the console tab? Thanks!

**Comment by sn0wcrashing** [comment]
> Hi, so I didn't managed to reproduce this bug. Could you open the dev tools (Ctrl+Shift+I) and show me if there's an error in the console tab? Thanks!


Sorry for the slowness in my response. 

Errors from Chrome in the Console are;

`Failed to load resource: the server responded with a status of 404 (Not Found)`

and 

`Failed to load resource: the server responded with a status of 409 (Conflict)`

![Chrome Console Error Screenshot](https://user-images.githubusercontent.com/30967099/170207891-02e2f858-810d-4570-a2f2-fe3242a9b8ec.png)

**Comment by abubaca4** [comment]
work with this nginx config
```
location = /transmission/ {
                return 301 /transmission/web/;
        }

        proxy_hide_header X-Frame-Options;

        location /transmission/ {
                proxy_http_version 1.1;
                proxy_set_header Connection "";
                proxy_read_timeout  300;
                proxy_set_header    Host                 $host;
                proxy_set_header    X-Forwarded-Proto    $scheme;
                proxy_set_header    X-Forwarded-Protocol $scheme;
                proxy_set_header    X-Real-IP            $remote_addr;
                proxy_pass_header   X-Transmission-Session-Id;
                proxy_set_header    X-Forwarded-Host     $host;
                proxy_set_header    X-Forwarded-Server   $host;
                proxy_set_header    X-Forwarded-For      $proxy_add_x_forwarded_for;
                proxy_pass          http://localhost:9091/transmission/;
        }
```
Try to set HA Proxy like this

**Comment by mateuszdrab** [comment]
Just discovered the app today @6c65726f79 thanks for developing it

I'm presuming the reason why it wouldn't work is if the SNI header isn't sent with TLS.

I too can't get the UI to work behind Haproxy.
I'll check if the header is sent and report back.

**Comment by mateuszdrab** [comment]
So it seems the issue was caused by a setting of Transmission, not the SNI as I initially suspected.
In the `settings.json` file located in `/usr/local/etc/transmission/home` (in case of theTrueNAS jail), you need to disable the `rpc-host-whitelist-enabled` setting.
Make sure to kill the transmission process first.

**Comment by 6c65726f79** [comment]
Thank you for sharing your solution @mateuszdrab, I hope this will help others.

By the way, thank you for the tip, it's greatly appreciated!

**Comment by mateuszdrab** [comment]
> Thank you for sharing your solution @mateuszdrab, I hope this will help others.
> 
> By the way, thank you for the tip, it's greatly appreciated!

I hope so too, since anyone behind a reverse proxy would probably be accessing via another hostname and by default RPC was being blocked. I couldn't even connect to it directly via hostname without the reverse proxy in the way.

Definitely deserve more tips buddy, I'm sure it took a while to make this app. The default web GUI for transmission is so ugly 😄

**Comment by taylorcox75** [comment]
https://github.com/6c65726f79/Transmissionic/pull/1374

**Comment by timothe** [comment]
Unfortunately this doesn't solve the issue for me. Transmission is on a Synology NAS behind a reverse proxy using a VPN. The 443 port is exposed and the Web UI is working. RPC communication seems to be working as well. But Transmissionic doesn't. I'm stuck on "Unable to reach host". Nothing shown in the developer tools.

**Comment by once375ml** [comment]
> Unfortunately this doesn't solve the issue for me. Transmission is on a Synology NAS behind a reverse proxy using a VPN. The 443 port is exposed and the Web UI is working. RPC communication seems to be working as well. But Transmissionic doesn't. I'm stuck on "Unable to reach host". Nothing shown in the developer tools.

I met the same situation and come here. I am sure you edited the "Server" section.

I also use reverse proxy for https connection. Transmissionic works if I don't touch any parameters in "Server" . Otherwise, I will also see "Unable to reach host".

![](https://s3.bmp.ovh/imgs/2023/11/30/1d7921fb7789c93d.png)

By the way, Where can I get the configuration file in client side? I want to make it "default.json" and upload to server side.

**Comment by remlio** [comment]
on https connections, the port must be 443, not 14000

**Comment by once375ml** [comment]
> on https connections, the port must be 443, not 14000

It's not the problem of port in synology reverse proxy. 

Transmissionic remembers the proxy domain, port and path locally, rather than the real domain, port and path from server.
